src/runtime_src/core/pcie/emulation/hw_em/alveo_shim/debug.cxx

```diff
@@ -309,14 +309,14 @@ namespace xclhwemhal2
       auto end_time = std::chrono::high_resolution_clock::now();
       if (std::chrono::duration_cast<std::chrono::seconds>(end_time - start_time).count() <= kMaxTimeToConnectSimulator) {
         
-        std::lock_guard<std::mutex> guard(mPrintMessagesLock);
         if (get_simulator_started() == false) 
           return;
-
-        dumpDeadlockMessages();
-        // Any status message found in parse log file?
-        lParseLog.parseLog();
-
+        if (true) { // This is for limiting the scope of lock
+          std::lock_guard<std::mutex> guard(mPrintMessagesLock);
```

Comment:
I don't care here, but FYI c++17 allows just `std:lock_guard lk{mutex}`, e.g. you don't have to need the template type.